src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,156 @@
+    exact to_sesq_form_apply_norm_le }
+end
+
+/-- The adjoint of a bounded operator from Hilbert space E to Hilbert space F. -/
+def adjoint : (E →L[𝕜] F) ≃ₗᵢ⋆[𝕜] (F →L[𝕜] E) :=
+linear_isometry_equiv.of_surjective
+{ norm_map' := adjoint'_norm,
+  ..adjoint' }
+(λ A, ⟨adjoint' A, adjoint'_adjoint' A⟩)
+
+local postfix `†`:1000 := adjoint
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_left (A : E →L[𝕜] F) (x : E) (y : F) : ⟪A† y, x⟫ = ⟪y, A x⟫ :=
+adjoint'_inner_left A x y
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_right (A : E →L[𝕜] F) (x : E) (y : F) : ⟪x, A† y⟫ = ⟪A x, y⟫ :=
+adjoint'_inner_right A x y
+
+/-- The adjoint is involutive -/
+@[simp] lemma adjoint_adjoint (A : E →L[𝕜] F) : A†† = A :=
+adjoint'_adjoint' A
+
+/-- The adjoint of the composition of two operators is the composition of the two adjoints
+in reverse order. -/
+@[simp] lemma adjoint_comp (A : F →L[𝕜] G) (B : E →L[𝕜] F) : (A ∘L B)† = B† ∘L A† :=
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  simp only [adjoint_inner_right, continuous_linear_map.coe_comp', function.comp_app],
+end
+
+lemma apply_norm_sq_eq_inner_adjoint_left (A : E →L[𝕜] E) (x : E) : ∥A x∥^2 = re ⟪(A† * A) x, x⟫ :=
+have h : ⟪(A† * A) x, x⟫ = ⟪A x, A x⟫ := by { rw [←adjoint_inner_left], refl },
+by rw [h, ←inner_self_eq_norm_sq _]
+
+lemma apply_norm_eq_sqrt_inner_adjoint_left (A : E →L[𝕜] E) (x : E) :
+  ∥A x∥ = real.sqrt (re ⟪(A† * A) x, x⟫) :=
+by rw [←apply_norm_sq_eq_inner_adjoint_left, real.sqrt_sq (norm_nonneg _)]
+
+lemma apply_norm_sq_eq_inner_adjoint_right (A : E →L[𝕜] E) (x : E) : ∥A x∥^2 = re ⟪x, (A† * A) x⟫ :=
+have h : ⟪x, (A† * A) x⟫ = ⟪A x, A x⟫ := by { rw [←adjoint_inner_right], refl },
+by rw [h, ←inner_self_eq_norm_sq _]
+
+lemma apply_norm_eq_sqrt_inner_adjoint_right (A : E →L[𝕜] E) (x : E) :
+  ∥A x∥ = real.sqrt (re ⟪x, (A† * A) x⟫) :=
+by rw [←apply_norm_sq_eq_inner_adjoint_right, real.sqrt_sq (norm_nonneg _)]
+
+/-- `E →L[𝕜] E` is a C⋆-algebra with the adjoint as the star operation. -/
+instance : has_star (E →L[𝕜] E) := ⟨adjoint⟩
+instance : has_involutive_star (E →L[𝕜] E) := ⟨adjoint_adjoint⟩
+instance : star_monoid (E →L[𝕜] E) := ⟨adjoint_comp⟩
+instance : star_ring (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_add adjoint⟩
+instance : star_module 𝕜 (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_smulₛₗ adjoint⟩
+
+lemma star_eq_adjoint (A : E →L[𝕜] E) : star A = A† := rfl
+
+instance : cstar_ring (E →L[𝕜] E) :=
+⟨begin
+  intros A,
+  rw [star_eq_adjoint],
+  refine le_antisymm _ _,
+  { calc ∥A† * A∥ ≤ ∥A†∥ * ∥A∥      : op_norm_comp_le _ _
+              ... = ∥A∥ * ∥A∥       : by rw [linear_isometry_equiv.norm_map] },
+  { rw [←sq, ←real.sqrt_le_sqrt_iff (norm_nonneg _), real.sqrt_sq (norm_nonneg _)],
+    refine op_norm_le_bound _ (real.sqrt_nonneg _) (λ x, _),
+    have := calc
+      re ⟪(A† * A) x, x⟫ ≤ abs ⟪(A† * A) x, x⟫  : re_le_abs _
+                  ...  ≤ ∥(A† * A) x∥ * ∥x∥     : abs_inner_le_norm _ _
```

Comment:
Could you please also extract this useful lemma
```lean
is_R_or_C.re ⟪x, y⟫ ≤ ∥x∥ * ∥y∥
```

Reply:
Done